how-to/how-to-use-k8s-with-cri-containerd-and-kata.md

```diff
@@ -0,0 +1,163 @@
+- Untrusted workload runtime: A runtime that will be used run untrusted workloads.
+
+#### Define the Kata runtime as `untrusted_workload_runtime` 
+
+Configure the Kata runtime for untrusted workload with the [config option](https://github.com/containerd/cri/blob/v1.0.0-rc.0/docs/config.md)
+`plugins.cri.containerd.untrusted_workload_runtime`.
+
+Unless configured otherwise, the default runtime is set to runc.
+```bash
+$ sudo mkdir -p /etc/containerd/
+$ cat << EOT | sudo tee /etc/containerd/config.toml
+[plugins]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.untrusted_workload_runtime]
+        runtime_type = "io.containerd.runtime.v1.linux"
+        runtime_engine = "/usr/bin/kata-runtime"
+EOT
+```
+
+### Configure Kubelet to use containerd
+
+In order to allow kubelet use containerd (using CRI interface)  configure the service to
+point to containerd socket.
+
+```bash
+# Configure k8s to use containerd
+$ sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
+$ cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+[Service]                                                 
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+EOF
+$ sudo systemctl daemon-reload
+```
+
+### Optional: Configure proxy
+
+If you are behind a proxy this script will configure your proxy for docker
+kubelet and containerd.
+
+```bash
+# Set proxys
+$services=(
+'kubelet'
+'containerd'
+'docker'
+)
+
+$ for s in "${services[@]}"; do 
+
+	service_dir="/etc/systemd/system/${s}.service.d/"
+	sudo mkdir -p ${service_dir}
+
+	cat << EOT | sudo tee "${service_dir}/proxy.conf"
+[Service]
+Environment="HTTP_PROXY=${http_proxy}"
+Environment="HTTPS_PROXY=${https_proxy}"
+Environment="NO_PROXY=${no_proxy}"
+EOT
+done
+$ sudo systemctl daemon-reload
+```
+
+###  Start Kubernetes with kubeadm 
+
+```bash
+# Mark sure contaienrd is up and running
+$ sudo systemctl restart containerd
+$ sudo systemctl status containerd
+
+# Prevent docker iptables rules confict with k8s pod communication
+$ sudo iptables -P FORWARD ACCEPT
+
+# Start cluster using kubeadm
+$ sudo kubeadm init --skip-preflight-checks \
+--cri-socket /run/containerd/containerd.sock --pod-network-cidr=10.244.0.0/16
+
+$ export KUBECONFIG=/etc/kubernetes/admin.conf
+
+$ sudo -E kubectl get nodes
+$ sudo -E kubectl get pods
+```
+
+# Install a pod network
+Install a pod network plugin is needed to allow pods communicate with each other.
+
+Install flannel plugging, by following the intructions in the section *Installing a pod network*
+from [Using kubeadm to Create a Cluster ](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/)
+guide.
+
+By default, the cluster will not schedule pods in the master node to allow that run:
+
+```
+$ sudo -E kubectl taint nodes --all node-role.kubernetes.io/master-
```

Comment:
If you want https://github.com/kata-containers/tests/blob/master/.ci/kata-doc-to-script.sh to include this in the script, please make this a `bash` code block.